src/Grpc/GrpcTransportTrait.php

```diff
@@ -29,22 +29,25 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-namespace Google\GAX;
+namespace Google\GAX\Grpc;
 
 use Google\Auth\ApplicationDefaultCredentials;
 use Google\Auth\CredentialsLoader;
 use Google\Auth\FetchAuthTokenCache;
 use Google\Auth\Cache\MemoryCacheItemPool;
+use Google\GAX\ValidationTrait;
 use Grpc\ChannelCredentials;
+use InvalidArgumentException;
 
-/**
- * A class that manages credentials for an API object using the Google Auth library
- */
-class GrpcCredentialsHelper
+trait GrpcTransportTrait
 {
     use ValidationTrait;
 
-    private $args;
+    private $grpcStub;
+    private $credentialsCallback;
+
+    // This value must be filled in by implementing classes
```

Comment:
This seems to be implementing inheritance, but in a trait. Why not make this a base class instead of a trait?

Reply:
As it uses a constructor and protected properties, it is a better fit for a parent class than a trait.